cmake/graph-compiler.cmake

```diff
@@ -0,0 +1,29 @@
+get_property(GRAPH_COMPILER_LIBS GLOBAL PROPERTY GRAPH_COMPILER_LIBS)
+if (NOT DEFINED GRAPH_COMPILER_LIBS)
+    include(FetchContent)
+
+    #FIXME: Replace the repository URL with the https://github.com/intel/graph-compiler
+    FetchContent_Declare(
+            GC
+            GIT_REPOSITORY https://github.com/AndreyPavlenko/graph-compiler.git
+            GIT_TAG pkg
+            FIND_PACKAGE_ARGS NAMES GraphCompiler
+    )
```

Comment:
@rengolin, could your guys do similar thing for tpp-mlir and probably for original llvm with specifying required version that is set inside tpp-mlir repository?

Reply:
I'd avoid building a strong relationship with either tpp-mlir and graph-compiler, as we may decide to have our own compiler reusing the same upstream technologies. If we get stuck on a specific compiler, we'll have less motivation to upstream and it will tie OV to whatever other project, which is not a good direction.

That's the main reason why we didn't build a robust build integration.